### ISSUE TITLE
None of the inputs have requires_grad=True. Gradients will be None

### ISSUE THREAD

**Issue by pspdada** [TinyLLaVA/TinyLLaVA_Factory#144]
I encountered the following phenomenon while using the `scripts/train/custom_finetune.sh` file for LoRA training: 
```
`/root/anaconda3/envs/tinyllava_factory/lib/python3.10/site-packages/torch/utils/checkpoint.py:31: UserWarning: None of the inputs have requires_grad=True. Gradients will be None
  warnings.warn("None of the inputs have requires_grad=True. Gradients will be None")`
```
Is this behavior normal?

My complete log:
```
[2024-12-04 16:38:56,254] [INFO] [launch.py:253:main] process 30693 spawned with command: ['/root/anaconda3/envs/tinyllava_factory/bin/python', '-u', 'tinyllava/train/custom_finetune.py', '--local_rank=1', '--deepspeed', './scripts/zero2.json', '--data_path', 'mydata/text_files/pokemon_blip_captions.json', '--image_folder', '/root/llm-project/TinyLLaVA_Factory/mydata', '--is_multimodal', 'True', '--conv_version', 'phi', '--mm_vision_select_layer', '-2', '--image_aspect_ratio', 'square', '--fp16', 'True', '--training_recipe', 'lora', '--tune_type_llm', 'lora', '--tune_type_vision_tower', 'frozen', '--tune_vision_tower_from_layer', '0', '--tune_type_connector', 'full', '--lora_r', '128', '--lora_alpha', '256', '--group_by_modality_length', 'False', '--pretrained_model_path', '/root/llm-project/TinyLLaVA_Factory/output/custom-finetune-TinyLLaVA-Phi-2-SigLIP-3.1B-lora', '--output_dir', '/root/llm-project/TinyLLaVA_Factory/output/custom-finetune-TinyLLaVA-Phi-2-SigLIP-3.1B-lora-2', '--num_train_epochs', '1', '--per_device_train_batch_size', '4', '--per_device_eval_batch_size', '4', '--gradient_accumulation_steps', '8', '--evaluation_strategy', 'no', '--save_strategy', 'steps', '--save_steps', '50000', '--save_total_limit', '1', '--learning_rate', '1e-4', '--weight_decay', '0.', '--warmup_ratio', '0.03', '--lr_scheduler_type', 'cosine', '--logging_steps', '1', '--tf32', 'False', '--model_max_length', '3072', '--gradient_checkpointing', 'True', '--dataloader_num_workers', '8', '--lazy_preprocess', 'True', '--report_to', 'tensorboard', '--tokenizer_use_fast', 'False', '--run_name', 'custom-finetune-TinyLLaVA-Phi-2-SigLIP-3.1B-lora']
[2024-12-04 16:38:59,546] [INFO] [real_accelerator.py:191:get_accelerator] Setting ds_accelerator to cuda (auto detect)
[2024-12-04 16:38:59,548] [INFO] [real_accelerator.py:191:get_accelerator] Setting ds_accelerator to cuda (auto detect)
[2024-12-04 16:39:03,421] [INFO] [comm.py:637:init_distributed] cdb=None
[2024-12-04 16:39:03,421] [INFO] [comm.py:637:init_distributed] cdb=None
[2024-12-04 16:39:03,421] [INFO] [comm.py:668:init_distributed] Initializing TorchBackend in DeepSpeed with backend nccl
Loading checkpoint shards: 100%|██████████████████████████████████████████████████████████████████████████████████████████████████████████████| 2/2 [00:30<00:00, 15.20s/it]
Loading checkpoint shards: 100%|██████████████████████████████████████████████████████████████████████████████████████████████████████████████| 2/2 [00:31<00:00, 15.60s/it]
Special tokens have been added in the vocabulary, make sure the associated word embeddings are fine-tuned or trained.
Special tokens have been added in the vocabulary, make sure the associated word embeddings are fine-tuned or trained.
2024-12-04 16:40:02,330 | INFO: Total Parameters: 3406160960, Total Trainable Parameters: 9507840
2024-12-04 16:40:02,345 | INFO: Trainable Parameters:
connector._connector.0.weight: 2949120 parameters
connector._connector.0.bias: 2560 parameters
connector._connector.2.weight: 6553600 parameters
connector._connector.2.bias: 2560 parameters
2024-12-04 16:40:04,138 | INFO: Added key: store_based_barrier_key:2 to store for rank: 0
2024-12-04 16:40:04,453 | INFO: Added key: store_based_barrier_key:2 to store for rank: 1
2024-12-04 16:40:04,453 | INFO: Rank 1: Completed store-based barrier for key:store_based_barrier_key:2 with 2 nodes.
2024-12-04 16:40:04,461 | INFO: Rank 0: Completed store-based barrier for key:store_based_barrier_key:2 with 2 nodes.
  0%|                                                                                                                                                | 0/13 [00:00<?, ?it/s]/root/anaconda3/envs/tinyllava_factory/lib/python3.10/site-packages/torch/utils/checkpoint.py:31: UserWarning: None of the inputs have requires_grad=True. Gradients will be None
  warnings.warn("None of the inputs have requires_grad=True. Gradients will be None")
/root/anaconda3/envs/tinyllava_factory/lib/python3.10/site-packages/torch/utils/checkpoint.py:31: UserWarning: None of the inputs have requires_grad=True. Gradients will be None
  warnings.warn("None of the inputs have requires_grad=True. Gradients will be None")
{'loss': 0.9132, 'grad_norm': 0.8190634250640869, 'learning_rate': 0.0001, 'epoch': 0.08}                                                                                   
{'loss': 0.9107, 'grad_norm': 0.926470935344696, 'learning_rate': 9.829629131445342e-05, 'epoch': 0.15}                                                                     
{'loss': 0.9457, 'grad_norm': 0.7495778799057007, 'learning_rate': 9.330127018922194e-05, 'epoch': 0.23}                                                                    
{'loss': 0.9811, 'grad_norm': 0.6585658192634583, 'learning_rate': 8.535533905932738e-05, 'epoch': 0.3}                                                                     
{'loss': 0.829, 'grad_norm': 0.7047551870346069, 'learning_rate': 7.500000000000001e-05, 'epoch': 0.38}                                                                     
{'loss': 0.8991, 'grad_norm': 0.802855372428894, 'learning_rate': 6.294095225512603e-05, 'epoch': 0.46}                                                                     
{'loss': 0.8786, 'grad_norm': 0.6734402775764465, 'learning_rate': 5e-05, 'epoch': 0.53}                                                                                    
{'loss': 0.8465, 'grad_norm': 0.6512352824211121, 'learning_rate': 3.705904774487396e-05, 'epoch': 0.61}                                                                    
{'loss': 0.9936, 'grad_norm': 0.6867278218269348, 'learning_rate': 2.500000000000001e-05, 'epoch': 0.69}                                                                    
{'loss': 0.9699, 'grad_norm': 0.6944343447685242, 'learning_rate': 1.4644660940672627e-05, 'epoch': 0.76}                                                                   
{'loss': 0.8196, 'grad_norm': 0.60866379737854, 'learning_rate': 6.698729810778065e-06, 'epoch': 0.84}                                                                      
{'loss': 0.9468, 'grad_norm': 0.7450130581855774, 'learning_rate': 1.70370868554659e-06, 'epoch': 0.91}                                                                     
{'loss': 0.9881, 'grad_norm': 0.806928277015686, 'learning_rate': 0.0, 'epoch': 0.99}                                                                                       
{'train_runtime': 117.5789, 'train_samples_per_second': 7.085, 'train_steps_per_second': 0.111, 'train_loss': 0.917064524613894, 'epoch': 0.99}                             
100%|███████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 13/13 [01:57<00:00,  9.03s/it]
[2024-12-04 16:42:16,481] [INFO] [launch.py:348:main] Process 30693 exits successfully.
/root/anaconda3/envs/tinyllava_factory/lib/python3.10/site-packages/transformers/integrations/peft.py:391: FutureWarning: The `active_adapter` method is deprecated and will be removed in a future version.
  warnings.warn(
[2024-12-04 16:42:30,497] [INFO] [launch.py:348:main] Process 30692 exits successfully.
```

My environment:
- `transformers` version: 4.40.1
- Platform: Linux-5.15.0-120-generic-x86_64-with-glibc2.35
- Python version: 3.10.15
- Huggingface_hub version: 0.26.3
- Safetensors version: 0.4.5
- Accelerate version: 0.27.2
- Accelerate config: not found
- PyTorch version (GPU?): 2.0.1+cu117 (True)
- Tensorflow version (GPU?): not installed (NA)
- Flax version (CPU?/GPU?/TPU?): not installed (NA)
- Jax version: not installed
- JaxLib version: not installed
- Using GPU in script?: 1
- Using distributed or parallel set-up in script?: None

**Comment by ZhangXJ199** [comment]
It is normal.

**Comment by pspdada** [comment]
Thanks for your reply!
